attestation-service/verifier/src/se/ibmse.rs

```diff
@@ -0,0 +1,318 @@
+// Copyright (C) Copyright IBM Corp. 2024
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{anyhow, Result};
+use log::{debug, warn};
+use openssl::{
+    encrypt::{Decrypter, Encrypter},
+    pkey::{PKey, Private, Public},
+    rsa::{Padding, Rsa},
+};
+use pv::{
+    attest::{
+        AdditionalData, AttestationFlags, AttestationItems, AttestationMeasAlg,
+        AttestationMeasurement, AttestationRequest, AttestationVersion,
+    },
+    misc::{open_file, read_certs, read_file},
+    request::{BootHdrTags, CertVerifier, HkdVerifier, ReqEncrCtx, Request, SymKeyType},
+    uv::ConfigUid,
+};
+use serde::{Deserialize, Serialize};
+use serde_json;
+use serde_with::{base64::Base64, serde_as};
+use std::{fs::File, io::Read};
+
+fn encrypt_measurement_key(key: &[u8], rsa_public_key: &PKey<Public>) -> Vec<u8> {
+    let mut encrypter = Encrypter::new(rsa_public_key).unwrap();
+    encrypter.set_rsa_padding(Padding::PKCS1).unwrap();
+
+    let buffer_len = encrypter.encrypt_len(key).unwrap();
+    let mut encrypted_hmac_key = vec![0; buffer_len];
+    let len = encrypter.encrypt(key, &mut encrypted_hmac_key).unwrap();
+    encrypted_hmac_key.truncate(len);
+
+    encrypted_hmac_key
+}
+
+fn decrypt_measurement_key(key: &[u8], rsa_private_key: &PKey<Private>) -> Vec<u8> {
+    let mut decrypter = Decrypter::new(rsa_private_key).unwrap();
+    decrypter.set_rsa_padding(Padding::PKCS1).unwrap();
+
+    let buffer_len = decrypter.decrypt_len(key).unwrap();
+    let mut decrypted_hmac_key = vec![0; buffer_len];
+    let decrypted_len = decrypter.decrypt(key, &mut decrypted_hmac_key).unwrap();
+    decrypted_hmac_key.truncate(decrypted_len);
+
+    decrypted_hmac_key
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct UserData {
+    #[serde_as(as = "Base64")]
+    image_btph: Vec<u8>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SeAttestationClaims {
+    #[serde_as(as = "Base64")]
+    cuid: ConfigUid,
+    #[serde_as(as = "Base64")]
+    user_data: Vec<u8>,
+    version: u32,
+    #[serde_as(as = "Base64")]
+    image_phkh: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    attestation_phkh: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    tag: [u8; 16],
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SeAttestationRequest {
+    #[serde_as(as = "Base64")]
+    request_blob: Vec<u8>,
+    measurement_size: u32,
+    additional_size: u32,
+    #[serde_as(as = "Base64")]
+    encr_measurement_key: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    encr_request_nonce: Vec<u8>,
+    #[serde_as(as = "Base64")]
+    image_hdr_tags: BootHdrTags,
+}
+
+impl SeAttestationRequest {
+    pub fn from_slice(request: &[u8]) -> Result<Self> {
+        Ok(serde_json::from_slice(request).unwrap())
+    }
+
+    pub fn from_string(request: &str) -> Result<Self> {
+        Ok(serde_json::from_str(request).unwrap())
+    }
```

Comment:
we have `serde_json::from_str` and `serde_json::from_slice` already ,thus no need to wrap the function again here.

Reply:
Fixed, thanks!